### PR TITLE
Boost: documenting the filters used in the plugin

### LINKED ISSUE #36250: [Boost] Revisit actions documented in Development Guide
I noticed that the list of hooks in the [DEVELOPMENT_GUIDE.md](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/boost/docs/DEVELOPEMENT_GUIDE.md) is outdated.

### PR DESCRIPTION
**Pull request by donnchawp**
The existing documentation covering the filters used in the plugin are woefully out of date.

Fixes [#36250](https://github.com/Automattic/jetpack/issues/36250)

## Proposed changes:

* Adding docs to docs/DEVELOPEMENT_GUIDE.md

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:
No testing required, but needs to be read and verified.

### CODE REVIEW

**Comment by donnchawp** [comment]
It looks like `jetpack_boost_module_enabled` has been completely removed. [It only exists now](https://github.com/search?q=repo%3AAutomattic%2Fjetpack+%22jetpack_boost_module_enabled%22&type=code) in tests and the documentation.

**Review by thingalon — COMMENTED** [review]
IMHO we're better off not documenting these in DEVELOPMENT_GUIDE.md, but instead ensuring that every hook that is called is documented in code from where it's called.

That will ensure the documentation is more likely kept up-to-date. This md file has been left behind and forgotten before.

I think we ought to document hooks inline with a docblock that explains what they're for, what they're passed, and (for filters) what they return.

Thoughts @pyronaur (as the one who wrote the original DEVELOPMENT_GUIDE.md) ?

**Comment by pyronaur** [comment]
@thingalon seeing is believing 😅 - the guide has fallen out of fashion multiple times. So yeah, I agree - documentation should probably go in code comments.

**Comment by haqadn** [comment]
Is this ready for another review/merge @donnchawp ?

**Comment by donnchawp** [comment]
@haqadn no, I decided other PRs were more urgent and put this aside for now. Should I close it?

**Comment by haqadn** [comment]
Since this is almost ready, lets finish this. @donnchawp

File: projects/plugins/boost/app/lib/Premium_Features.php [hunk 1]
```diff
@@ -48,6 +48,13 @@ public static function get_features() {
 		$features = array();
 		// Prepare a list of features after applying jetpack_boost_has_feature_* filter for each feature.
 		foreach ( $all_features as $feature ) {
+			/**
+			 * Filter the feature list to disable some features.
```
Comment (haqadn) [inline projects/plugins/boost/app/lib/Premium_Features.php]: It's not filtering the feature list, the filter is applies to features individually. So, I think it should say something like: Filter the availability of a feature.

File: projects/plugins/boost/app/lib/class-viewport.php [hunk 2]
```diff
@@ -149,6 +149,15 @@ public static function get_viewport_size() {
...
+		 * Filter the viewport size. This is used for testing.
+		 *
+		 * @param array `$data`: array containing the width and height of the view port
+		 * @param string `$cookie_name`: Name of the viewport cookie
+		 * @param string `$cookie_value`: value of the viewport cookie
+		 *
+		 * @since   1.0.0
+		 */
 		return apply_filters( 'jetpack_boost_viewport_size', $data, self::VIEWPORT_COOKIE, $cookie_value );
 	}
```
Comment (haqadn) [inline projects/plugins/boost/app/lib/class-viewport.php]: I couldn't find any usage of the `Viewport` class. I think we can just get rid of this file and the corresponding test.

File: projects/plugins/boost/app/lib/minify/Concatenate_CSS.php [hunk 3]
```diff
@@ -192,10 +229,28 @@ public function do_items( $handles = false, $group = false ) {
 					$style_tag = "<link rel='stylesheet' id='$css_id' href='$href' type='text/css' media='$media' />";
 				}
 
+				/**
+				 * Filter the style loader HTML tag for page optimize.
+				 *
+				 * @param string $tag style loader tag
```
Comment (haqadn) [inline projects/plugins/boost/app/lib/minify/Concatenate_CSS.php]: To match with the filter usage:

```suggestion
				 * @param string $style_tag style loader tag
```

File: projects/plugins/boost/app/lib/minify/Concatenate_CSS.php [hunk 4]
```diff
@@ -192,10 +229,28 @@ public function do_items( $handles = false, $group = false ) {
...
+				 * @since   1.0.0
+				 */
 				$style_tag = apply_filters( 'page_optimize_style_loader_tag', $style_tag, $handles, $href, $media );
 
-				// Allow manipulation of the stylesheet tag.
-				// For example - making it deferred when using with Critical CSS.
+				/**
+				 * Filter the stylesheet tag. For example: making it deferred when using Critical CSS.
+				 *
+				 * @param string $tag stylesheet tag
```
Comment (haqadn) [inline projects/plugins/boost/app/lib/minify/Concatenate_CSS.php]:
```suggestion
				 * @param string $style_tag stylesheet tag
```

**Review by haqadn — CHANGES_REQUESTED** [review]
I have suggested some corrections. The DEVELOPER_GUIDE.md file is outdated in a few more ways:

- The "Module architectural overview 101" is outdated in more ways. I think we can get rid of this section for now.
- The table of content is out of date and we need to remove the removed sections from it.

File: projects/plugins/boost/app/lib/Premium_Features.php (hunk 1 above)
Comment (donnchawp) [reply projects/plugins/boost/app/lib/Premium_Features.php]: Thanks.

File: projects/plugins/boost/app/lib/class-viewport.php (hunk 2 above)
Comment (donnchawp) [reply projects/plugins/boost/app/lib/class-viewport.php]: Done

**Review by haqadn — DISMISSED** [review]
👍🏼

**Review by haqadn — APPROVED** [review]

**Pull request merged**